pkg/apis/core/helper/seed.go

```diff
@@ -0,0 +1,152 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package helper
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/utils/ptr"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+)
+
+// TaintsHave returns true if the given key is part of the taints list.
+func TaintsHave(taints []core.SeedTaint, key string) bool {
+	for _, taint := range taints {
+		if taint.Key == key {
+			return true
+		}
+	}
+	return false
+}
+
+// TaintsAreTolerated returns true when all the given taints are tolerated by the given tolerations.
+func TaintsAreTolerated(taints []core.SeedTaint, tolerations []core.Toleration) bool {
+	if len(taints) == 0 {
+		return true
+	}
+	if len(taints) > len(tolerations) {
+		return false
+	}
+
+	tolerationKeyValues := make(map[string]string, len(tolerations))
+	for _, toleration := range tolerations {
+		v := ""
+		if toleration.Value != nil {
+			v = *toleration.Value
+		}
+		tolerationKeyValues[toleration.Key] = v
+	}
+
+	for _, taint := range taints {
+		tolerationValue, ok := tolerationKeyValues[taint.Key]
+		if !ok {
+			return false
+		}
+		if taint.Value != nil && *taint.Value != tolerationValue {
+			return false
+		}
+	}
+
+	return true
+}
+
+// SeedSettingSchedulingVisible returns true if the 'scheduling' setting is set to 'visible'.
+func SeedSettingSchedulingVisible(settings *core.SeedSettings) bool {
+	return settings == nil || settings.Scheduling == nil || settings.Scheduling.Visible
+}
+
+// SeedSettingTopologyAwareRoutingEnabled returns true if the topology-aware routing is enabled.
+func SeedSettingTopologyAwareRoutingEnabled(settings *core.SeedSettings) bool {
+	return settings != nil && settings.TopologyAwareRouting != nil && settings.TopologyAwareRouting.Enabled
+}
+
+var scheme *runtime.Scheme
+
+func init() {
+	scheme = runtime.NewScheme()
+	utilruntime.Must(core.AddToScheme(scheme))
+	utilruntime.Must(gardencorev1beta1.AddToScheme(scheme))
+}
```

Comment:
`init` func is usually placed at the top of the file, right after the imports.

See examples in:
- https://github.com/gardener/gardener/blob/d82db8674c240691f9276c6775dc04e573bb8c05/cmd/gardenlet/app/options.go#L24-L35
- https://github.com/gardener/gardener/blob/41a694b024144465bd4f02f39d6d1c3493549c2f/pkg/api/extensions/accessor_test.go#L25-L32